indexer/indexer_tests/editable_map_object_test.cpp

```diff
@@ -191,37 +191,52 @@ UNIT_TEST(EditableMapObject_GetNamesDataSource)
 
   TEST_EQUAL(namesDataSource.names.size(), 9, ("All names except the default should be pushed into "
                                            "data source plus empty mandatory names"));
-  TEST_EQUAL(namesDataSource.mandatoryNamesCount, 3,
-             ("Mandatory names count should be equal == Mwm languages + user`s language"));
+  TEST_EQUAL(namesDataSource.mandatoryNamesCount, 4,
+             ("Mandatory names count should be equal to Mwm languages + user`s language"));
   TEST_EQUAL(namesDataSource.names[0].m_code, GetLangCode("de"),
-             ("Deutsch name should be first as first language in Mwm"));
+             ("German must be the first because it is first in the list of languages for MWM"));
   TEST_EQUAL(namesDataSource.names[1].m_code, GetLangCode("fr"),
-             ("French name should be second as second language in Mwm"));
-  TEST_EQUAL(namesDataSource.names[2].m_code, GetLangCode("ko"),
-             ("Korean name should be third because user`s language should be followed by Mwm languages"));
+             ("German must be the second because it is second in the list of languages for MWM"));
+  TEST_EQUAL(namesDataSource.names[2].m_code, GetLangCode("en"),
+             ("English name should be placed after Mwm languages"));
+  TEST_EQUAL(namesDataSource.names[3].m_code, GetLangCode("ko"),
+             ("Korean should be fourth because the user’s langue must be followed by English."));
 
   {
     vector<int8_t> nativeMwmLanguages = {GetLangCode("de"), GetLangCode("fr")};
 
     auto const namesDataSource =
         EditableMapObject::GetNamesDataSource(emo.GetName(), nativeMwmLanguages, GetLangCode("fr"));
     TEST_EQUAL(namesDataSource.names.size(), 9,
-               ("All names except the default should be pushed into data source"));
-    TEST_EQUAL(namesDataSource.mandatoryNamesCount, 2,
-               ("Mandatory names count should be equal == Mwm languages + user`s language. "
-                "Excluding repetiton"));
+               ("All names + empty mandatory names should be pushed into "
+                "the data source, except the default one."));
+    TEST_EQUAL(namesDataSource.mandatoryNamesCount, 3,
+               ("Mandatory names count should be equal to MWM languages + "
+                "english language + user`s language. Excluding repetiton"));
```

Comment:
the English language. Language names are used either without the world "language" and no article, or with the world "language" and with the definite article. I.e. "English" or "The English language"